ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/Command.java

```diff
@@ -119,6 +160,73 @@ public String toString() {
     return "Command{"
         + "statement='" + statement + '\''
         + ", overwriteProperties=" + overwriteProperties
+        + ", version=" + version
         + '}';
   }
+
+  @VisibleForTesting
+  static class VersionChecker {
+
+    private static final RangeMap<ArtifactVersion, Range<ArtifactVersion>> DB_COMPAT_MATRIX =
```

Comment:
that is indeed much simpler - I will go down this path, but I think we need to start with version `0` so that this isn't backwards incompatible unnecessarily